### PR TITLE
[BB-3618] Implement page structure

### PR DESCRIPTION
**Pull request by farhaanbukhsh**
We are trying to migrate to the 2 columns new page structure for the console. This way the UI has an overlay on the sidebar nav to make all the adjustments possible.

**JIRA tickets**: [BB-3618](https://tasks.opencraft.com/browse/BB-3618)

**Discussions**: [Public GitLab Ticket](https://gitlab.com/opencraft/dev/opencraft/-/issues/701)

**Dependencies**:  ~~This is still dependent on https://github.com/open-craft/opencraft/pull/701 but initially set of reviews can be done irrespective. Later we can integrate the studio page.~~

**Screenshots**: 
image

image

image

~~**Sandbox URL**: TBD - sandbox is being provisioned.~~

~~**Merge deadline**: "None" if there's no rush, "ASAP" if it's critical, or provide a specific date if there is one.~~

**Testing instructions**:

1. Install and run the OCIM interface
2. You need to comment the following lines to bypass validation
```
diff --git a/registration/api/v2/serializers.py b/registration/api/v2/serializers.py
index 5e1f28d4..08ee02fa 100644
--- a/registration/api/v2/serializers.py
+++ b/registration/api/v2/serializers.py
@@ -362,8 +362,8 @@ class OpenEdXInstanceConfigSerializer(serializers.ModelSerializer):
         is_new_instance = self.instance is None
         is_changed = not is_new_instance and self.instance.subdomain != self.initial_data.get("subdomain")
 
-        if is_new_instance or is_changed:
-            validate_available_subdomain(value)
+        # if is_new_instance or is_changed:
+        #     validate_available_subdomain(value)
 
         return value

```
3. You need to manually visit the link http://localhost:3000/newconsole
4. Click on the Logo Suboption in Accordion you will see the layout change
5.  For testing logo update page you need to revert few changes from https://github.com/open-craft/opencraft/pull/714/commits/fe0a0034d9e79c9369a32d215a094534cc5b6a41
6. The logo update page is dummy and it's just to have a concept implementation here/
7. We need to check the transition is happening when we click Course (Studio) you will be taken to new dashboard and any other option will take you to old board. 

**Author notes and concerns**:

1. We don't need to verify the LOGO component since it won't be used as-is. The point of integrating that component was to check and build the functionality for the same.
2. As mentioned when the dependent PR will get merge there will be a link from the old console page to a new one, this is done temporarily till the whole page gets created. We want to ship new features every sprint.
3. The whole point of the PR is to get the structure ready and integrate the Course (Studio) page in the new console. This way we will have the foundation of future work.

**Reviewers**
- [ ] @0x29a

### CODE REVIEW

**Comment by antoviaque** [comment]
@farhaanbukhsh @0x29a Thanks for working on this!

Btw, to double-check - you have in mind to get Ali & Cassie to review the changes as part of the PR review, right?

**Comment by farhaanbukhsh** [comment]
> @farhaanbukhsh @0x29a Thanks for working on this!
> 
> Btw, to double-check - you have in mind to get Ali & Cassie to review the changes as part of the PR review, right?
I was thinking if @0x29a  does another pass I can deploy this on stage and give Ali and Cassie the access :) if you have anything otherwise let me know :) thanks

**Review by 0x29a — CHANGES_REQUESTED** [review]
One of two main points of my review: diff is hard to read. It consists mostly of code copied from other components, but that's unavoidable given the approach we've chosen. Changes themselves look mostly fine to me.

Are we going to merge this in master and deploy on prod? In that case, I'm a bit worried. In some of Fixate documents this task is marked as high risk, meaning that it can prevent the user from completing his task. Also, I don't feel very comfortable mixing old and new designs on the same page and letting users face the interface that is under construction.

If we're going to keep this PR open and just merge in other dependent PRs, can't we instead create a branch like `new-console`, run a second frontend under `new.console.opencraft.com` domain and update it until all large UI tasks are done? This way we can avoid meaningless diffs during work on next console rework tasks and as a result avoid many accidental typos / errors. Also, we won't need to do renaming of anything with `new` prefix when epic will be completed.

I've seen an approach with a new frontend on a separate domain used by ProtonMail, Jenkins Blue Ocean and a bunch of other services.

What do you think, @giovannicimolin and @farhaanbukhsh?

**Comment by farhaanbukhsh** [comment]
> One of two main points of my review: diff is hard to read. It consists mostly of code copied from other components, but that's unavoidable given the approach we've chosen. Changes themselves look mostly fine to me.

Right and I am sorry for the heavy diff :(

> 
> Are we going to merge this in master and deploy on prod? In that case, I'm a bit worried. In some of Fixate documents this task is marked as high risk, meaning that it can prevent the user from completing his task. Also, I don't feel very comfortable mixing old and new designs on the same page and letting users face the interface that is under construction.
> 
> If we're going to keep this PR open and just merge in other dependent PRs, can't we instead create a branch like `new-console`, run a second frontend under `new.console.opencraft.com` domain and update it until all large UI tasks are done? This way we can avoid meaningless diffs during work on next console rework tasks and as a result avoid many accidental typos / errors. Also, we won't need to do renaming of anything with `new` prefix when epic will be completed.
> 


Our initial plan was to have a different branch altogether and then  merge it, but then `Xavier` pointed out that we should not hold back anything and ship features often as per the scrum model, you can read the [discussion](https://tasks.opencraft.com/browse/BB-3372?focusedCommentId=187012&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-187012)

Hope this is convincing enough :smile: 



> I've seen an approach with a new frontend on a separate domain used by ProtonMail, Jenkins Blue Ocean and a bunch of other services.

something we can plan for the next iteration maybe.

> 
> What do you think, @giovannicimolin and @farhaanbukhsh?

File: frontend/src/newConsole/actions.ts [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+  DeploymentInfoModel,
+  DeploymentNotificationModel
+} from 'console/models';
+import {
+  ThemeSchema,
+  StaticContentOverrides,
+  ToggleStaticContentPages
+} from 'ocim-client';
+import { NOTIFICATIONS_LIMIT, ROUTES } from 'global/constants';
+import { sanitizeErrorFeedback } from 'utils/string_utils';
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/newConsole/actions.ts:16]: :+1: to this comment.

File: frontend/src/newConsole/components/newConsolePage/styles.scss [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    font-size: 20px;
+  }
+
+  h3 {
+   font-family: $chivo-font;
+   font-size: 15px;
+ }
+
+  .new-console-page-container {
+    background-color:rgba(109, 156, 174, 0.3);
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/newConsole/components/newConsolePage/styles.scss:40]: :+1: to this comment.

File: frontend/src/newConsole/components/newCustomizationSideMenu/displayMessages.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  accordionTheme: {
+    defaultMessage: 'Theme',
+    description: 'The name of the theme menu in the sidebar accordion.'
+  },
+  linkPreviewColors: {
+    defaultMessage: 'Preview & colors',
+    description: 'Menu name.'
+  },
+  linkLogos: {
+    defaultMessage: 'Logos',
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/newConsole/components/newCustomizationSideMenu/displayMessages.ts:11]: :+1:

File: frontend/src/newConsole/reducers.ts [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+  let activeInstanceId: number;
+
+  switch (action.type) {
+    case Actions.Types.CLEAR_CONSOLE_DATA:
+      return initialConsoleState;
+    case Actions.Types.CLEAR_ERROR_MESSAGE:
+      return update(state, {
+        activeInstance: {
+          feedback: { $unset: [action.field] }
+        }
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/newConsole/reducers.ts:19]: :+1: ^

File: frontend/src/newConsole/components/newLogos/displayMessages.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      "If unset, OpenCraft's logo will be used.",
+    description: ''
+  },
+  uploadFavicon: {
+    defaultMessage: 'Upload Favicon',
+    description: ''
+  }
+};
+
+export default messages;
```
[the remark this answers is not in the document]
Comment (giovannicimolin) [reply frontend/src/newConsole/components/newLogos/displayMessages.ts:32]: :+1: ^

**Comment by giovannicimolin** [comment]
>>Are we going to merge this in master and deploy on prod? In that case, I'm a bit worried. In some of Fixate documents this task is marked as high risk, meaning that it can prevent the user from completing his task. Also, I don't feel very comfortable mixing old and new designs on the same page and letting users face the interface that is under construction.
If we're going to keep this PR open and just merge in other dependent PRs, can't we instead create a branch like new-console, run a second frontend under new.console.opencraft.com domain and update it until all large UI tasks are done? This way we can avoid meaningless diffs during work on next console rework tasks and as a result avoid many accidental typos / errors. Also, we won't need to do renaming of anything with new prefix when epic will be completed.

>Our initial plan was to have a different branch altogether and then merge it, but then Xavier pointed out that we should not hold back anything and ship features often as per the scrum model, you can read the discussion
Hope this is convincing enough smile

We already discussed this before, and it isn't the optimal UX to show users, but it's what allows us to ship features to production on every sprint. CCing @antoviaque since this is a product-ish discussion.

File: frontend/src/newConsole/components/newLogos/LogoSidbar.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
```
Comment (arjunsinghy96) [inline frontend/src/newConsole/components/newLogos/LogoSidbar.tsx:1]: Typo in filename: `LogoSidbar` -> `LogoSidebar`

Comment (farhaanbukhsh) [reply frontend/src/newConsole/components/newLogos/LogoSidbar.tsx:1]: Thanks

**Review by 0x29a — CHANGES_REQUESTED** [review]
@farhaanbukhsh, I see `package-lock.json` file is heavily modified, but no new packages are added. Why are we including it? If we don't add new packages, lock file should be unchanged.

Also, I see that history was rewritten after the last review, changes are significant, but there are only a few new commits. So, it's impossible for me as reviewer to decouple new changes from previously added basing on commits history and diff and do a proper review pass. Usually it's a good tone to avoid rebasing untill code is approved to merge.

Am I right that changes I have to review are in https://github.com/open-craft/opencraft/pull/714/commits/2c9dce5c34c7fa554a661ac9b826a3b3e88eba54 commit?

**Comment by farhaanbukhsh** [comment]
> @farhaanbukhsh, I see `package-lock.json` file is heavily modified, but no new packages are added. Why are we including it? If we don't add new packages, lock file should be unchanged.
> 
I was looking for the tests to pass and they were failing without pack.json changes, I will remove this.

> Also, I see that history was rewritten after the last review, changes are significant, but there are only a few new commits. So, it's impossible for me as reviewer to decouple new changes from previously added basing on commits history and diff and do a proper review pass. Usually it's a good tone to avoid rebasing untill code is approved to merge.
> 

I was rebasing it on master Demid since those changes were affecting the Ux review. Will keep this in mind.

> Am I right that changes I have to review are in [2c9dce5](https://github.com/open-craft/opencraft/commit/2c9dce5c34c7fa554a661ac9b826a3b3e88eba54) commit?

yes you are right :)

**Comment by 0x29a** [comment]
>I was looking for the tests to pass and they were failing without pack.json changes

@farhaanbukhsh, hmm, does CI fail as well without lock file?

>>Am I right that changes I have to review are in 2c9dce5 commit?

>yes you are right :)

Okay then, I'm approving once the lock file thing is sorted. New UI looks good, thanks for doing this. :+1:

**Comment by farhaanbukhsh** [comment]
@0x29a  tests are passing :)

**Review by 0x29a — APPROVED** [review]
:+1:

- [x] I tested that implemented interface works, matched mock-up, checked that new code isn't throwing errors.
- [x] I read through the code

**Pull request merged**